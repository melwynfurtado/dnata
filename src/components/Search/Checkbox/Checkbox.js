import React from 'react'

const Checkbox = ({ id, value, handleChange, children }) => (
  <div className="form-check">
    <input className="form-check-input" type="checkbox" id={id} value={value} onChange={handleChange}/>
    <label className="form-check-label" htmlFor={id}>{ children }</label>
  </div>  
)

export default Checkbox