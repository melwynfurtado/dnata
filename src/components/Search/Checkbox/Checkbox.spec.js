import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Checkbox from './Checkbox'

describe('Checkbox Component', () => {
  let handleOnChangeMock
  let Component

  beforeEach(() => {
    handleOnChangeMock = jest.fn()
    Component = <Checkbox id="5star" value="5" handleChange={handleOnChangeMock}>5 stars</Checkbox>
  })

  it('renders correctly', () => {
    const tree = renderer
                  .create(Component)
                  .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls handleChange onchange', () => {
    const wrapper = shallow(Component)
    wrapper.find('.form-check-input').simulate('change')
    expect(handleOnChangeMock).toHaveBeenCalled()
  })
})