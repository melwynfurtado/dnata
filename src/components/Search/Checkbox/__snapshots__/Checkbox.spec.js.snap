// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component renders correctly 1`] = `
<div
  className="form-check"
>
  <input
    className="form-check-input"
    id="5star"
    onChange={[MockFunction]}
    type="checkbox"
    value="5"
  />
  <label
    className="form-check-label"
    htmlFor="5star"
  >
    5 stars
  </label>
</div>
`;
